index.rst

```diff
@@ -2,24 +2,180 @@
+
+Exceptional Cases
+-----------------
+
+.. Since part of Telescope and Site Software scope is to deliver a CI/CD system for the observatory control software, this system
+
+Although a substantial number of requirements can be verified with the use of our CI system, we acknowledge that some of them will inevitably demand a more individualized approach and, sometimes, even a more involved process demanding additional data analysis and/or a more descriptive verification methodology.
+
+Ideally, these cases would be addressed by either developing a customized SAL Script to drive the system through the required test steps or, in the more extreme conditions that might require interactive steps, a Jupyter notebook.
+
+Furthermore, any detailed data analysis/methodology should be included in a technical note to be incorporated into the requirement verification artifacts.
+
+.. , since we execute (or plan to) these tests continuously, we can verify that the system continues to fullfil requirements at every iteration.
+
+.. Because these functionalities are (mostly) taken from the requirements pool, it means that each new feature is generally accompanied by a requirement verification test.
+.. Nevertheless, tests comes in a variety of different shapes and forms and, most importantly, sometimes it is not possible to fully exercise a given requirement in a testing environment.
+
+.. _verification-artifacts:
+
+Verification artifacts
+======================
+
+In general, successfully executing the testing suite from start to finish gives us confidence that the system is operating as expected.
+Nevertheless, the SAL Scripts used during these tests are only designed to fail execution if some critical condition happen (e.g. a component unexpectedly went to FAULT state), and are not designed to verify all the details of every single component involved in the process.
+
+For example, the ``set_summary_state`` SAL Script is designed to execute state transitions and is commonly used in operations to enable components, recover from fault states, etc.
+They are executed as an initial step during testing to enable all components and, while they verify the components do transition successfully to the desired state, they **do not** check that the CSCs publish all the required events.
+This type of verification need to be done a posteriori and included in verification artifacts.
+
+These artifacts are human readable reports that contain enough information for a reader with some knowledge about the system can understand what is being verified and how.
+For instance, in the example above, the report could show that a particular CSC in the STANDBY state will transition to the DISABLED state after receiving a ``start`` command and will also publish a series of events in the process, with a particular expected payload.
+Furthermore, if a component is supposed to publish certain events and fail to do so, the test reports will mark the condition as a failure.
+
```

Comment:
Add a tie-in sentence to the next section.
We utilize the Robot Framework software suite(?) to help generate, manage and evaluate these reports.